### PR TITLE
HCM Refactor

### PR DESCRIPTION
**Pull request by Flova**
## Proposed changes

- Add consistent typing
- Fix some errors
- Remove unused code
- Fix idle spinning in sim (100% CPU + spam in downstream nodes like the behavior)
- Reduce walking and HCM rates to sim speed
- Fix kick param

### CODE REVIEW

File: bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py [hunk 1]
```diff
@@ -13,6 +10,7 @@ class WaitForIMUStartup(AbstractActionElement):
 
     def __init__(self, blackboard, dsd, parameters=None):
         super().__init__(blackboard, dsd, parameters)
+        self.blackboard: HcmBlackboard
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py:13]: Don't you want to create `AbstractHCMActionElement` like for the Decisions

File: bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait.py [hunk 2]
```diff
@@ -23,18 +22,23 @@ def __init__(self, blackboard, dsd, parameters=None):
...
+                             + self.blackboard.node.get_clock().now().seconds_nanoseconds()[1]/1e9):
             self.pop()
 
 class WaitNoReevaluate(Wait):
     def perform(self, reevaluate=False):
         self._dsd.set_do_not_reevaluate()
-        if self.time < float(self.blackboard.get_clock().now().seconds_nanoseconds()[0] + self.get_clock().now().seconds_nanoseconds()[1]/1e9):
+        if self.time < float(self.blackboard.node.get_clock().now().seconds_nanoseconds()[0] \
+                              + self.blackboard.node.get_clock().now().seconds_nanoseconds()[1]/1e9):
             self.pop()
```
Comment (jaagut) [inline bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait.py:44]: This class is not relevant anymore

**Review by jaagut — APPROVED** [review]

File: bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py [hunk 3]
```diff
@@ -134,7 +137,7 @@ def diag_cb(self, msg: DiagnosticArray):
             elif "//Pressure" in status.name:
                 self.pressure_diag_error = status.level == DiagnosticStatus.ERROR or status.level == DiagnosticStatus.STALE
 
-    def last_kick_feedback_callback(self, msg):
+    def last_kick_feedback_callback(self, msg: PlayAnimation.Feedback):
         self.last_kick_feedback = self.node.get_clock().now()
```
[the remark this answers is not in the document]
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/hcm_blackboard.py:141]: It is not available

File: bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py (hunk 1 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait_for.py:13]: Good point, but not that important at the moment imo..

File: bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait.py (hunk 2 above)
Comment (Flova) [reply bitbots_hcm/bitbots_hcm/hcm_dsd/actions/wait.py:44]: I agree

**Pull request merged**